lib/nokogiri_handler.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class NokogiriHandler
+  class << self
+    # See "set of space-separated tokens" in the HTML5 spec.
+    WHITE_SPACE = /[ \x09\x0A\x0C\x0D]+/
+
+    def link_rel_include(token_list, token)
```

Comment:
I think this could be useful in other places as well. And it is not specific to `rel`, is it?

If I am correct, how about renaming this to something more generic like `case_independent_include` or similar?

Reply:
In terms of the HTML spec, the attribute contains a case-insensitive list of space-separated DOM tokens. The specific definition of “space” in this context is represented by the regexp. There aren't a lot attributes defined in this way, so I chose a name reflecting the current use-case for the function.

DOM tokens are not always case-insensitive, so I initially created a whole bunch of generic functions for converting to lower-case, splitting into space-separated tokens etc., but then I decided that this would be way too generic for this specific use-case. Mastodon is not a library, so we can easily change the method name or split into separate functions if the need arises.

That was my reasoning, but I would be happy to rename etc. if you think that would be better.

Reply:
I think another use-case would be the `name` attribute of `<meta>` tags (see a couple of lines below your change in `LinkDetailsExtractor`). If I understand the spec correctly, the name is also case insensitive.

Reply:
It is indeed case-insensitive, but it is not defined as a space-separated list of tokens, so I don't think this function is a perfect fit (though it will probably do the job except in a few edge cases). In #31357 (overlapping with this PR) I have proposed a `casecmp` handler function.

Reply:
Let us agree than once again that naming is hard :slightly_smiling_face: 

I still believe that this method could have utility elsewhere, but you are of course correct: Without a concrete example this point is kind of moot.

So let us leave it as is. Thanks for your patience.